### DISCUSSION TITLE
Transforming Left Joins into Foreign Key Relationships in Row File

### DISCUSSION THREAD

**Discussion by EdwinThomas11** [serenity-is/Serenity#7088] [Help]
[ConnectionKey("Default"), Module("Default"), TableName("[Table1]")]
[LeftJoin("SP", "Table2", "SP.[Table1Id] = t0.[Id] AND SP.[IsActive] = 1")]
[LeftJoin("SD", “Table3, "SD.[Table1Id] = t0.[Id] AND SD.[IsActive] = 1")]

In my row file, I have a left join structured as above. Now, I need to transform it into a foreign key, linking to the property.

[ForeignKey(typeof(Table2Row), "Id"), LeftJoin("SP")]
public Int64? Table2Id

@VictorTomaili @volkanceylan  please help here.

**Comment by VictorTomaili** [comment]
Use expression `[SP].Id` and please do not mention us for a reply.

**Reply by EdwinThomas11** [reply]
Certainly, I mentioned it because I anticipated a prompt resolution for this matter.

My intention is to eliminate the following left join clauses:

[LeftJoin("SP", "Table2", "SP.[Table1Id] = t0.[Id] AND SP.[IsActive] = 1")]
[LeftJoin("SD", "Table3", "SD.[Table1Id] = t0.[Id] AND SD.[IsActive] = 1")

I aim to manage this through a property, as the presence of two records in my child table is causing duplicate entries to appear on the list page. I need the TotalCount property to be exact in my table1 (Parent table) as per list criteria.

Thanks for the quick help.

**Reply by VictorTomaili** [reply]
Use expression and write a sub query there. The answer is the same.

**Reply by volkanceylan** [reply]
If you are a premium customer use StartSharp repo or mail, if not you may only expect community support, open source is not unlimited free support, so stop mentioning anyone
